draft-banks-quic-performance.md

```diff
@@ -110,6 +113,64 @@ When a client uses a unidirectional stream to request a response payload from
 the server, the server opens a new unidirectional stream to send the requested
 data.  If no data is requested by the client, the server need take no action.
 
+## Unreliable Datagrams
+
+For endpoints that support the QUIC Unreliable Datagram extension, datagrams may
+also be used.  The datagrams are used similarly, but with a few minor differences
+to account for their unreliable nature:
+
+- The client encodes a flow ID and the response size in every datagram sent.
+
+When a server receives a datagram, it does the following:
+
+- The server decodes the flow ID and the response size.
+- If the server has not received a datagram with a new flow ID, it then responds
+  with its own datagrams with matching flow ID.  It sends datagrams up to the
+  requested response size.
+
+**Note** - This design requires the server to maintain connection-wide state.
+This could become a DoS vector.  To avoid this, the client SHOULD use
+monotomically increasing flow ID values and the server may trivially keep track
+of the highest flow ID received to decide on whether it should respond.  This
+solution is not perfect as it can result in reordered datagrams from being
+ignored.
+
+### Encoding Flow ID and Response Size
+
+Every datagram sent by the client uses the first 8 bytes of the payload to
+encode a 64-bit unsigned integer in network byte order to indicate the flow ID,
+and the following 8 bytes to encode a 64-bit unsigned integer in network byte
+order to indicate the total length of data the client wishes the server to
+respond with.  This can be larger that can fit into a single datagram.  In that
+case, the server should continue to send datagrams in response until the total
+length is reached.
+
+To account for possible loss of datagrams, the client encodes the total response
+length for that flow ID in every datagram it sends.  Then, the server needs to
+only respond to the first (if any) datagram it receives.
+
+An encoded value of zero for the response size is perfectly legal, and a value
+of MAX_UINT64 (0xFFFFFFFFFFFFFFFF) is practically used to indicate an unlimited
+server response, but there is no way for the client to cancel receipt of this
+data besides closing the connection.
+
+On the server side, any datagram that received but does not contain enough
+payload for the flow ID and response size should be ignored.
+
+## Error Codes
+
+Error codes on connections are unused in this protocol and are ignored on
+receipt.
```

Comment:
This text seems a little loose. I think you're trying to say that no application error codes are defined and received values have no meaning. That's fine. But the current text could be misread as transport errors also have no meaning, which would be very odd.